### PR TITLE
Dev feat509 hashing v0.8

### PR DESCRIPTION
**Pull request by serializingme**
New version of the pull request #2102 fixing a major bug where the detection function wasn't comparing the file hash (MD5, SHA1 or SHA256) with the correct in memory hashes (e.g., comparing the file MD5 hash with the SHA256 hashes). This was clear because it was matching files not having any of the specified hashes (in the case of a negated rule), when they did.

### CODE REVIEW

File: src/util-detect-file-hash.c [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    if (s->file_flags & FILE_SIG_NEED_MD5 && file->flags & FILE_MD5) {
+        match = HashMatchHashTable(filehash->hash, file->md5, sizeof(file->md5));
+    }
+    else if (s->file_flags & FILE_SIG_NEED_SHA1 && file->flags & FILE_SHA1) {
+        match = HashMatchHashTable(filehash->hash, file->sha1, sizeof(file->sha1));
+    }
+    else if (s->file_flags & FILE_SIG_NEED_SHA256 && file->flags & FILE_SHA256) {
+        match = HashMatchHashTable(filehash->hash, file->sha256, sizeof(file->sha256));
+    }
+
```
Comment (serializingme) [inline src/util-detect-file-hash.c]: For those curious, the problem with #2102 was in the previous three conditions, where it wasn't checking to what hashing algorithm the signature refereed to.

**Comment by inliniac** [comment]
Can you rename the commit 'Common code added' to be more descriptive on the subject line? E.g. 'file hashing: add common code'. That way while browsing commits/git blame it's more obvious what the commit is about at first glance.

**Comment by serializingme** [comment]
Done, I have changed the messages to better align with the Suricata team commit format.

**Comment by pevma** [comment]
I rerun the my previous tests - it looks all matching is good now on sha1/sha256/md5 when using white/black listings of the checksums. Thanks!

**Comment by inliniac** [comment]
Can you do a rebase?

**Pull request closed without merging**